supportedBackends/spark/src/main/scala/cromwell/backend/impl/spark/SparkJobExecutionActor.scala

```diff
@@ -160,17 +158,18 @@ class SparkJobExecutionActor(override val jobDescriptor: BackendJobDescriptor,
       log.debug("{} Creating bash script for executing command: {}", tag, command)
       // TODO: we should use shapeless Heterogeneous list here not good to have generic map
       val attributes: Map[String, Any] = Map(
-        SparkCommands.AppMainClass -> runtimeAttributes.appMainClass,
+        SparkCommands.AppMainClass -> runtimeAttributes.appMainClass.getOrElse(""),
         SparkCommands.Master -> sparkMaster,
         SparkCommands.ExecutorCores -> runtimeAttributes.executorCores,
-        SparkCommands.ExecutorMemory -> runtimeAttributes.executorMemory.to(MemoryUnit.GB).amount.toLong,
+        SparkCommands.ExecutorMemory -> runtimeAttributes.executorMemory.toString.replaceAll("\\s","").toLowerCase,
+        SparkCommands.AdditionalArgs -> runtimeAttributes.additionalArgs.getOrElse(""),
         SparkCommands.SparkAppWithArgs -> command.get,
         SparkCommands.DeployMode -> sparkDeployMode
       )
 
       val sparkSubmitCmd = cmds.sparkSubmitCommand(attributes)
       val sparkCommand = if (isClusterMode) {
-        sparkSubmitCmd.concat(" %s %s".format("&>", SubmitJobJson.format(sparkDeployMode)))
+        sparkSubmitCmd.concat(" %s %s 2>&1".format(">", SubmitJobJson.format(sparkDeployMode)))
```

Comment:
why all this format trickery rather than the usual interpolation? and only thing really seems to need to be interpolated.